### DISCUSSION TITLE
Name and icon together as View title

### DISCUSSION THREAD

**Discussion by edo2313** [home-assistant/frontend#11190] [Other feature requests]
I would like to be able to set names and icons together as a view title so the tabs on tab are like
" Bedroom *bed icon* "
Is this possible?

**Comment by iantrich** [comment]
Not currently but an option to do so sounds reasonable.

**Comment by balloob** [comment]
It's not part of the material design guidelines. Nor do I think that it is a common use case to warrant diverging from the material design guidelines for it.

**Comment by iantrich** [comment]
It is part of the spec: https://material.io/design/components/tabs.html#

**Comment by edo2313** [comment]
Yeah it is part of the guidelines... I think that if people have the option they will use it, you can get the idea of the view immediately if you see an icon

**Comment by thomasloven** [comment]
Kind of an edge-case here...

![skarmavbild 2018-12-30 kl 00 29 17](https://user-images.githubusercontent.com/1299821/50542998-01064c00-0bca-11e9-822b-e665c7ccbd62.png)

**Comment by edo2313** [comment]
Well, I don't think that someone will put icons only on some tabs and not on others...

**Comment by iantrich** [comment]
Our optional setting would be at the top-level and apply to all tabs. So someone COULD override our default behavior but we would be following the spec

**Comment by zsarnett** [comment]
We would have to default an icon for each tab then

**Comment by thomasloven** [comment]
> Well, I don't think that someone will put icons only on some tabs and not on others...

I can't tell if you're serious or not...
Of course someone would!
**I** would!

**Comment by iantrich** [comment]
> We would have to default an icon for each tab then

We could do that, but don't think it is necessary. Would just be more work for @thomasloven to then have to set `icon: mdi:na` for each view explicitly when he overrides it 😜

**Comment by iantrich** [comment]
Additionally, we already allow the spec to be broken
![image](https://user-images.githubusercontent.com/1287159/65075134-e6445500-d95b-11e9-86b4-1a5e94d2dc06.png)

**Comment by iantrich** [comment]
If all tabs have icons and text, display both
Else if all tabs have icons, display icons
Else display text

**Comment by iantrich** [comment]
Will likely hold off on this until we switch to mwc tabs: https://material-components.github.io/material-components-web-components/demos/tabs.html
